taichi/ui/backends/vulkan/vertex.h

```diff
@@ -21,9 +23,19 @@ struct Vertex {
   };
   vec3 pos;
   vec3 normal;
+  // FIXME: tex_coord
   vec2 texCoord;
   vec4 color;
 };
 
+enum class VboAttribes {
+  kAll,
+  kPos,
+  kPosNormal,
+  kPosNormalUv,
+};
```

Comment:
Isn't this still a bit restrictive? 
For example, a user might choose to provides ndarrays for only Pos and Color. In this case, the VBO would still contain Normal and UV, and we'd run into the same problem that we have now.
Would it be better to have something like
```cpp
enum VertexAttribs : int
{
    Pos = 0b1,
    Normal = 0b10,
    Uv = 0b100,
    Color = 0b1000,
}
inline VertexAttribs operator|(VertexAttribs a, VertexAttribs b)
{
    return static_cast<VertexAttribs>(static_cast<int>(a) | static_cast<int>(b));
}
bool hasAttrib(VertexAttribs a, VertexAttribs attrib){
   return static_cast<int>(a) &  static_cast<int>(attrib);
}
```

Reply:
Ah yes, will change to this one, thanks!